services/kafka/service.go

```diff
@@ -71,6 +75,7 @@ func (w *writer) Open() {
 	statsMap.Set(statWriteMessageCount, writeMessages)
 
 	w.ticker = time.NewTicker(time.Second)
+	w.tickerDone = make(chan struct{}, 1)
```

Comment:
this doesn't need to have a `1)` since its only ever used by closing.  It isn't that big a deal though.